lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -118,36 +157,83 @@ def call_to_persisted?(node)
           node.send_type? && node.method?(:persisted?)
         end
 
+        def assignable_node(node)
+          assignable = node_or_passed_block(node)
+          while node
+            node = hash_parent(node) || array_parent(node)
+            assignable = node if node
+          end
+          assignable
+        end
+
+        # for some reason passed blocks are parents of send rather than children
+        # but the first child of the block is the receiver of the block argument
+        def node_or_passed_block(node)
+          block = node.parent
```

Comment:
Are we confident that `node` passed as an argument is a `send`, `super`, or `yield` node? If so, you can use the node extension and do:

```ruby
node.block_node || node
```

Reply:
ooo, thank you